### PR TITLE
fix block longhaul test

### PR DESCRIPTION
**Pull request by dangula**
[smoke only]

### CODE REVIEW

File: tests/framework/utils/k8s_helper.go [hunk 1]
```diff
@@ -688,15 +688,12 @@ func (k8sh *K8sHelper) GetExternalRGWServiceURL(storeName string, namespace stri
...
-	logger.Debugf("svc : %v", svc)
-	logger.Debugf("err : %v", err)
-	if err != nil {
-		if svc.Size() == 1 {
-			return true
-		}
+	_, err := k8sh.Clientset.Services(namespace).Get("rook-api", opts)
+	if err == nil {
+		return true
+
```
Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: did `gofmt` not remove this blank line? A blank line before a bracket is good to clean up in any case

File: tests/framework/utils/mysql_helper.go [hunk 2]
```diff
@@ -102,6 +115,7 @@ func (h *MySQLHelper) TableRowCount() (count int) {
 			panic(err)
 		}
 	}
+	rows.Close()
```
Comment (travisn) [inline tests/framework/utils/mysql_helper.go]: should this be a defer statement at the beginning of the method?

File: tests/framework/utils/mysql_helper.go [hunk 3]
```diff
@@ -133,5 +147,6 @@ func (h *MySQLHelper) DeleteRandomRow() sql.Result {
 	if err != nil {
 		panic(err)
 	}
+	rows.Close()
```
Comment (travisn) [inline tests/framework/utils/mysql_helper.go]: defer?

File: tests/longhaul/block_on_k8s_test.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"sync"
 	"testing"
 
+	"github.com/coreos/pkg/capnslog"
```
Comment (travisn) [inline tests/longhaul/block_on_k8s_test.go]: all our tests are for k8s now, so i'd suggest you just name this file `block_test.go`

File: tests/longhaul/block_on_k8s_test.go [hunk 5]
```diff
@@ -44,7 +49,7 @@ func (s *K8sBlockLongHaulSuite) SetupSuite() {
 	assert.Nil(s.T(), err)
 
 	s.installer = installer.NewK8sRookhelper(s.kh.Clientset)
-	if !s.kh.IsRookInstalled() {
+	if !s.kh.IsRookInstalled("rook") {
```
Comment (travisn) [inline tests/longhaul/block_on_k8s_test.go]: do we not have a const for the default namespace? (and better not to use "rook" in the tests so we make sure we work in different namespaces from dev env)

File: tests/longhaul/block_on_k8s_test.go [hunk 6]
```diff
@@ -196,18 +203,21 @@ func (s *K8sBlockLongHaulSuite) dbOperation(i int) {
...
 	//delete Data
 	s.db.DeleteRandomRow()
+	s.db.SelectRandomData(20)
 
 }
 func (s *K8sBlockLongHaulSuite) TearDownSuite() {
 	s.db.CloseConnection()
-	s.kh.ResourceOperation("delete", s.mysqlAppPath)
-	s.storageClassOperation("mysql-pool", "delete")
+	//s.kh.ResourceOperation("delete", s.mysqlAppPath)
```
Comment (travisn) [inline tests/longhaul/block_on_k8s_test.go]: why not clean these up?

File: tests/longhaul/block_on_k8s_test.go (hunk 5 above)
Comment (dangula) [reply tests/longhaul/block_on_k8s_test.go]: that was in a different package. Add them here too

File: tests/longhaul/block_on_k8s_test.go (hunk 6 above)
Comment (dangula) [reply tests/longhaul/block_on_k8s_test.go]: We want to repeat the test with the same setup, reuse the same volume basically.

File: tests/framework/utils/k8s_helper.go (hunk 1 above)
Comment (travisn) [reply tests/framework/utils/k8s_helper.go]: ok thanks. It's fixed? It still looks like the space is there

File: tests/framework/utils/mysql_helper.go [hunk 7]
```diff
@@ -95,17 +95,17 @@ func (h *MySQLHelper) InsertRandomData() sql.Result {
 func (h *MySQLHelper) SelectRandomData(limit int) *sql.Rows {
 	query := fmt.Sprintf("SELECT * FROM  LONGHAUL ORDER BY RAND() LIMIT %d", limit)
 	rows, err := h.DB.Query(query)
+	defer rows.Close()
```
Comment (travisn) [inline tests/framework/utils/mysql_helper.go]: will `rows` be nil when `err != nil`? If so, the `defer` might need to go after the err check

File: tests/longhaul/block_test.go [hunk 8]
```diff
@@ -24,7 +24,9 @@ func TestK8sBlockLongHaul(t *testing.T) {
 }
 
 var (
-	logger = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
+	logger               = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
+	defaultNamespace     = "default"
+	defaultRookNamespace = "test-rook"
```
Comment (travisn) [inline tests/longhaul/block_test.go]: if you're going to redefine this const, i'd suggest a different namespace such as `longhaul`

File: tests/longhaul/block_on_k8s_test.go (hunk 6 above)
Comment (travisn) [reply tests/longhaul/block_on_k8s_test.go]: what test reuses them? This means another test has a dependency on this one?

File: tests/framework/utils/k8s_helper.go (hunk 1 above)
Comment (dangula) [reply tests/framework/utils/k8s_helper.go]: ok, fixed now.

File: tests/longhaul/block_on_k8s_test.go (hunk 6 above)
Comment (dangula) [reply tests/longhaul/block_on_k8s_test.go]: For the long-haul test, we don't want to delete the volume after each run, we want to set up a volume and use it for multiple hours or days. The tests deploy rook under a unique namespace, so other regression tests should not be affected. Also, we have called out that long-haul test doesn't clean up the test data in the readme.

File: tests/longhaul/block_test.go (hunk 8 above)
Comment (dangula) [reply tests/longhaul/block_test.go]: updated.

File: tests/framework/utils/mysql_helper.go (hunk 7 above)
Comment (dangula) [reply tests/framework/utils/mysql_helper.go]: yes, good catch, moving defer after err check

File: tests/longhaul/block_on_k8s_test.go (hunk 6 above)
Comment (travisn) [reply tests/longhaul/block_on_k8s_test.go]: ok, would be great to add a comment here about why they aren't cleaned up. And if there is no intention of uncommenting these lines, seems like you could delete them.

Comment (dangula) [reply tests/longhaul/block_on_k8s_test.go]: Added a comment to the test calling out no clean up

**Pull request merged**